thoth-export-server/src/bibtex/bibtex_thoth.rs

```diff
@@ -144,106 +118,118 @@ impl TryFrom<Work> for BibtexThothEntry {
                 "Missing Publication Date".to_string(),
             ));
         }
-        let mut author_list = vec![];
-        let mut editor_list = vec![];
+
         let mut contributions = work.contributions;
         // WorkQuery should already have retrieved these sorted by ordinal, but sort again for safety
         contributions.sort_by(|a, b| a.contribution_ordinal.cmp(&b.contribution_ordinal));
-        for contribution in contributions {
-            if contribution.main_contribution {
-                if work.work_type == WorkType::EDITED_BOOK {
-                    if contribution.contribution_type == ContributionType::EDITOR {
-                        editor_list.push(contribution.full_name);
-                    }
-                } else if contribution.contribution_type == ContributionType::AUTHOR {
-                    author_list.push(contribution.full_name);
-                }
+        let (author, mut editor) = extract_authors_and_editors(contributions)?;
+
+        let shorttitle = work.subtitle.as_ref().map(|_| work.title.clone());
+        let mut booktitle = None;
+        let mut chapter = None;
+        let mut pages = None;
+        let mut entry_type = "book".to_string();
+        if work.work_type == WorkType::BOOK_CHAPTER {
+            entry_type = "inbook".to_string();
+            // Inbook type does not allow both author and editor – unset editor if both are present
```

Comment:
As discussed, we don't think there's a difference between `inbook` and `book` behaviour - should either do this for both types or not at all (and the suggestion was to let the user handle the warning, i.e. "not at all")